### ISSUE TITLE
No recommendations seen though notification says "Duration Based Recommendations Available"

### ISSUE THREAD

**Issue by chandrams** [kruize/autotune#770]
No recommendations seen though notification says "Duration Based Recommendations Available". This is seen when the 96th result data is posted and recommendations is seen from 97th.

```
 "recommendations": {
                            "notifications": [
                                {
                                    "type": "info",
                                    "message": "Duration Based Recommendations Available"
                                }
                            ],
                            "data": {
                                "2023-04-23T11:46:12.152Z": {
                                    "duration_based": {
                                        "short_term": {
                                            "pods_count": 0,
                                            "confidence_level": 0.0,
                                            "notifications": [
                                                {
                                                    "type": "info",
                                                    "message": "There is not enough data available to generate a recommendation."
                                                }
                                            ]
                                        },
                                        "medium_term": {
                                            "pods_count": 0,
                                            "confidence_level": 0.0,
                                            "notifications": [
                                                {
                                                    "type": "info",
                                                    "message": "There is not enough data available to generate a recommendation."
                                                }
                                            ]
                                        },
                                        "long_term": {
                                            "pods_count": 0,
                                            "confidence_level": 0.0,
                                            "notifications": [
                                                {
                                                    "type": "info",
                                                    "message": "There is not enough data available to generate a recommendation."
                                                }
                                            ]
                                        }
                                    }
                                },
```

**Comment by chandrams** [comment]
Recommendations json for reference
[recommendations.zip](https://github.com/kruize/autotune/files/11302977/recommendations.zip)

**Comment by kgaikwad** [comment]
Hello @chandrams and @dinogun, 
Just an thought - Along with notification type, will it be possible to introduce something like messageType as key with keyword as value? In my opinion, it is helpful instead of message string comparison.
For any minor change in string, condition will start to fail.
Thoughts?

**Comment by bharathappali** [comment]
Currently if kruize has exact 24 hrs of data it is not able to give out recommendations due to a boundary condition issue which was mentioned by @chandrams and @kusumachalasani .

@msvinaykumar found that the issue is with the way Recommendation engine is getting the start time.

Kruize is having the check for sum of duration minutes at the start of iteration, so if the list has exact number of entries as it can't iterate more the check doesn't happen in the last iteration. Hence kruize is not generating the recommendation if it has exact entries for a duration. Which will be fixed in #827

**Comment by dinogun** [comment]
Fixed in #827
